### PR TITLE
chore: update operator-sdk version to 1.32

### PR DESCRIPTION
**Pull request by anandrkskd**
**What type of PR is this?**
/kind chore

**What does this PR do / why we need it**:
This PR upgrades the operator-sdk version to v1.32 

**Have you updated the necessary documentation?**

* [ ] Documentation update is required by this PR.
* [ ] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #?

**Test acceptance criteria**:

* [ ] Unit Test
* [ ] E2E Test

**How to test changes / Special notes to the reviewer**:

Download the 1.32 operator-sdk binary, run make operator-sdk also move this to your $PATH. Then run the makefile targets to verify the upgrade.

### CODE REVIEW

File: config/crd/bases/pipelines.openshift.io_gitopsservices.yaml [hunk 1]
```diff
@@ -1,9 +1,11 @@
+
 ---
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.14.0
+    controller-gen.kubebuilder.io/version: v0.4.1
```
Comment (svghadi) [inline config/crd/bases/pipelines.openshift.io_gitopsservices.yaml:7]: Looks like your local version of controller-gen is old. Can you check this?

Comment (anandrkskd) [reply config/crd/bases/pipelines.openshift.io_gitopsservices.yaml]: updated

**Comment by saumeya** [comment]
In [test/e2e/suite_test.go](https://github.com/anandrkskd/gitops-operator/blob/16ac342b2ded4a5f62b8ea4619dd3269c92f7041/test/e2e/suite_test.go#L96) and [test/nondefaulte2e/suite_test.go](https://github.com/anandrkskd/gitops-operator/blob/16ac342b2ded4a5f62b8ea4619dd3269c92f7041/test/nondefaulte2e/suite_test.go#L80) , please update the CRDDirectoryPaths to []string{filepath.Join("..", "..", "config", "crd", "bases")},

**Comment by anandrkskd** [comment]
> We also need to pull in the latest `ArgoCD` and `NotificationsConfiguration` CRD from https://github.com/argoproj-labs/argocd-operator/tree/master/config/crd/bases to https://github.com/redhat-developer/gitops-operator/tree/master/config/crd/bases and update argocd-operator commit to latest in go.mod.

I am planning to do the `go.mod` changes in next part of operator-sdk version upgrade 1.35

**Review by saumeya — APPROVED** [review]
LGTM

**Comment by svghadi** [comment]
> I am planning to do the go.mod changes in next part of operator-sdk version upgrade 1.35

Sounds good. Looks like we also need to update operator-sdk version in openshift-ci configs. I see one reference here : https://github.com/redhat-developer/gitops-operator/blob/6cf2a8e31d153a5585ee699722f894404b295dc7/openshift-ci/build-root/Dockerfile#L6 

@varshab1210 are there any other places where operator-sdk version  is hardcoded, maybe in [openshift/release](https://github.com/openshift/release) repo?

**Pull request merged**